code/game/objects/structures/guillotine.dm

```diff
@@ -0,0 +1,230 @@
+		msg += "The blade is raised, ready to fall, and"
+
+		if (blade_sharpness >= GUILLOTINE_DECAP_MIN_SHARP)
+			msg += " looks sharp enough to decapitate without any resistance."
+		else
+			msg += " doesn't look particularly sharp. Perhaps a whetstone can be used to sharpen it."
+	else
+		msg += "The blade is hidden inside the stocks."
+
+	if (LAZYLEN(buckled_mobs))
+		msg += "<br/>"
+		msg += "Someone appears to be strapped in. You can help them out, or you can harm them by activating the guillotine."
+
+	to_chat(user, msg)
+
+	return msg
+
+/obj/structure/guillotine/attack_hand(mob/user)
+	if (blade_status == GUILLOTINE_BLADE_MOVING)
+		return
+
+	if (blade_status == GUILLOTINE_BLADE_DROPPED)
+		blade_status = GUILLOTINE_BLADE_MOVING
+		icon_state = "guillotine_raise"
+		addtimer(CALLBACK(src, .proc/raise_blade), GUILLOTINE_ANIMATION_LENGTH)
+		return
+
+	if (blade_status == GUILLOTINE_BLADE_RAISED)
+		if (LAZYLEN(buckled_mobs))
+			if (user.a_intent == INTENT_HARM)
+				user.visible_message("<span class='warning'>[user] begins to pull the lever!</span>",
+					                 "<span class='warning'>You begin to the pull the lever.</span>")
+				if (do_after(user, GUILLOTINE_ACTIVATE_DELAY, target = src) && blade_status == GUILLOTINE_BLADE_RAISED)
+					blade_status = GUILLOTINE_BLADE_MOVING
+					icon_state = "guillotine_drop"
+					addtimer(CALLBACK(src, .proc/drop_blade), GUILLOTINE_ANIMATION_LENGTH - 2) // Minus two so we play the sound and decap faster
+			else
+				var/mob/living/carbon/human/H = buckled_mobs[1]
+
+				if (H)
+					H.regenerate_icons()
+
+				unbuckle_all_mobs()
+		else
+			blade_status = GUILLOTINE_BLADE_MOVING
+			icon_state = "guillotine_drop"
+			addtimer(CALLBACK(src, .proc/drop_blade), GUILLOTINE_ANIMATION_LENGTH)
+
+/obj/structure/guillotine/proc/raise_blade()
+	blade_status = GUILLOTINE_BLADE_RAISED
+	icon_state = "guillotine_raised"
+
+/obj/structure/guillotine/proc/drop_blade()
+	if (buckled_mobs.len && blade_sharpness)
+		var/mob/living/carbon/human/H = buckled_mobs[1]
+
+		if (H)
+			var/obj/item/bodypart/head/head = H.get_bodypart("head")
+
+			if (head)
+				playsound(src, 'sound/weapons/bladeslice.ogg', 100, 1)
+				if (blade_sharpness >= GUILLOTINE_DECAP_MIN_SHARP || head.brute_dam >= 100)
+					head.dismember()
+					H.regenerate_icons()
+					unbuckle_all_mobs()
+					kill_count += 1
+
+					var/blood_overlay = "bloody"
+
+					if (kill_count == 2)
+						blood_overlay = "bloodier"
+					else if (kill_count > 2)
+						blood_overlay = "bloodiest"
+
+					blood_overlay = "guillotine_" + blood_overlay + "_overlay"
+					cut_overlays()
+					add_overlay(mutable_appearance(icon, blood_overlay))
+
+					// The crowd is pleased
+					// The delay is to making large crowds have a longer laster applause
+					var/delay_offset = 0
+					for(var/mob/M in viewers(src, 7))
+						var/mob/living/carbon/human/C = M
+						if (ishuman(M))
+							addtimer(CALLBACK(C, /mob/.proc/emote, "clap"), delay_offset * 0.3)
```

Comment:
if you're willing to do extension work this should give them a positive moodlet

Reply:
I'm not familiar with moodlets, is there a go-to proc for increasing or adding positive moodlet?

Reply:
afraid not, you'd have to create a moodlet datum and send the appropriate signal.

this is fine as is though so no worries.